src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,199 @@
+            IEnumerable<ISymbol> members = semanticModel.LookupSymbols(position, type);
+            members = members.Where(m => m.CanBeReferencedByName &&
+                IsFieldOrReadableProperty(m) &&
+                !m.IsImplicitlyDeclared);
+
+            // Filter out those members that have already been typed
+            var alreadyTestedMembers = GetTestedMembers(semanticModel.SyntaxTree, position, cancellationToken);
+            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name) &&
+                m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+
+            foreach (var untestedMember in untestedMembers)
+            {
+                context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
+                    displayText: untestedMember.Name.EscapeIdentifier(),
+                    insertionText: null,
+                    symbols: ImmutableArray.Create(untestedMember),
+                    contextPosition: token.GetLocation().SourceSpan.Start,
+                    rules: s_rules));
+            }
+        }
+
+        private static bool IsFieldOrReadableProperty(ISymbol symbol)
+        {
+            if (symbol.IsKind(SymbolKind.Field))
+            {
+                return true;
+            }
+
+            if (symbol.IsKind(SymbolKind.Property) && !((IPropertySymbol)symbol).IsWriteOnly)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static readonly CompletionItemRules s_rules = CompletionItemRules.Create(enterKeyRule: EnterKeyRule.Never);
+
+        internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
+            => CompletionUtilities.IsTriggerCharacter(text, characterPosition, options) || text[characterPosition] == ' ';
+
+        private static SyntaxToken TryGetOpenBraceOrCommaInPropertyPattern(
+            Document document, SemanticModel semanticModel, int position, CancellationToken cancellationToken)
+        {
+            var tree = semanticModel.SyntaxTree;
+            if (tree.IsInNonUserCode(position, cancellationToken))
+            {
+                return default;
+            }
+
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            token = token.GetPreviousTokenIfTouchingWord(position);
+
+            if (!token.IsKind(SyntaxKind.CommaToken, SyntaxKind.OpenBraceToken))
+            {
+                return default;
+            }
+
+            if (token.Parent.IsKind(SyntaxKind.PropertySubpattern) == false ||
+                token.Parent.Parent.IsKind(SyntaxKind.PropertyPattern) == false)
```

Comment:
== false and || feels weird....  why not just ```if token.parent.iskind() && token.parent.parent.iskind() ? token : default```?

Reply:
the parent of a PropertySubpattern can also be a DeconstructionPattern:
```c#
(1, 2) { P1: 1, P2: 2 }
```